### PR TITLE
Add toast reporter for Karma

### PR DESCRIPTION
**Pull request by mplewis**
Provides helpful status messages to devs when their builds pass/fail in the background:

pass
fail
none

Ready for review.

# TODO

- [x] Smoke test locally
- [ ] Verify that this actually works on Linux
- [x] Test the config to ensure it doesn't break CI
- [ ] Cut a new version for npm

### CODE REVIEW

File: lib/gusto-karma.config.js [hunk 1]
```diff
@@ -5,6 +5,7 @@ var paths = require('./utils/path');
 var ROOT_DIR = paths.ROOT_DIR;
 var SPEC_DIR = paths.SPEC_DIR;
 var browserConfig = require('./utils/browsers');
+var toastPlugin = require('./utils/toastReporter');
```
Comment (nikhilmat) [inline lib/gusto-karma.config.js]: this is the wrong path

**Comment by nikhilmat** [comment]
@mplewis please check that this works locally (the filepath would have broken things)

you can install packages via filepath to check it locally before pushing a new version

**Comment by nikhilmat** [comment]
https://yarnpkg.com/lang/en/docs/cli/add/

**Comment by mplewis** [comment]
Thanks for the instructions, I've never tested an npm package locally before. Will test and report back.

**Comment by mplewis** [comment]
@nikhilmat: I used `yarn link` to link `gusto-karma` into `zenpayroll` for testing. I'm hitting the following crashes around Underscore and Babel:

```sh
$ yarn run test
yarn run v0.19.1
$ gusto-karma test
Running Gusto-Karma Tests..

START:
Environment has changed (node_modules or configuration was updated).
HardSourceWebpackPlugin will reset the cache and store a fresh one.

ERROR in ../gusto-karma/lib/karma.test_runner.js
Module not found: Error: Cannot resolve module 'babel' in /Users/matt.lewis/workspace/gusto-karma/lib
 @ ../gusto-karma/lib/karma.test_runner.js 1:0-36
10 03 2017 14:19:24.433:INFO [karma]: Karma v1.3.0 server started at http://localhost:9876/
10 03 2017 14:19:24.435:INFO [launcher]: Launching browser Chrome_with_debugging with unlimited concurrency
10 03 2017 14:19:24.453:INFO [launcher]: Starting browser Chrome
10 03 2017 14:19:25.619:INFO [Chrome 57.0.2987 (Mac OS X 10.12.3)]: Connected on socket /#XDlAsPQUf9NlhtUWAAAA with id 25269221
Chrome 57.0.2987 (Mac OS X 10.12.3) ERROR
  Uncaught Error: Cannot find module "lib/extensions/underscore"
  at /Users/matt.lewis/workspace/gusto-karma/lib/karma.test_runner.js:78
Chrome 57.0.2987 (Mac OS X 10.12.3) ERROR
  Uncaught Error: Cannot find module "lib/extensions/underscore"
  at /Users/matt.lewis/workspace/gusto-karma/lib/karma.test_runner.js:78

Finished in 0.937 secs / 0 secs @ 14:19:26 GMT-0800 (PST)

SUMMARY:
✔ 0 tests completed
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

Any ideas?

**Comment by nikhilmat** [comment]
@mplewis yarn link has some weird dependency resolution issues because it uses a symlink. for example, you'd normally have access to babel in the node_modules directory where packages are installed, but since its symlinking to the source directory, it has no access to a babel package unless you were to install locally (along with all its other dependencies).

try using yarn add an a file path, since that will copy the files in the proper directory

**Comment by mplewis** [comment]
Thanks for the tips. I had issues using `yarn add` in dev with Yarn's cache. The trick was:

```sh
rm -rf node_modules
yarn cache clear
yarn install
yarn run test
```

This works locally via `yarn run test` and `IN_CI=TRUE yarn run test`.

I'll try to find someone with a Linux box to test on. What other testing would you like me to try before you're comfortable with this PR?

**Comment by nikhilmat** [comment]
@mplewis let's merge this in, and then open a PR in ZP that updates the package. I think as long as it works in CI, thats fine and if you're opting out that should do the trick.

@zp-alfred accept

https://gustohq.atlassian.net/wiki/display/ED/Publish+a+private+NPM+Package

**Comment by zp-alfred** [comment]
Accepted by nikhilmat!

**Comment by mplewis** [comment]
Reviewing the publish private NPM instructions now – thanks. Do I need to bump the version number of gusto-karma in this PR?

**Comment by mplewis** [comment]
I'll wait until Monday to publish and pull this package into zenpayroll, so no time pressure.

**Comment by nikhilmat** [comment]
Yes you'd need to bump the version to not overwrite the existing package at the version that is checked in

**Comment by mplewis** [comment]
Version bumped. Just waiting on someone with Linux to try this out before I push to NPM Private.

**Comment by nikhilmat** [comment]
@mplewis i think its probably not imperative, the main thing I wanted to check was that it doesn't break CI. I think matan/aron were the last people to use docker in development and have since gone back to developing on their local machine

**Comment by mplewis** [comment]
ok! I will merge if you are OK with the version bump to 2.1.0.

**Comment by nikhilmat** [comment]
yep

**Pull request closed without merging**